### ISSUE TITLE
Is it possible to add multiple items to grid at once?

### ISSUE THREAD

**Issue by ClleaR** [tiberiuzuld/angular-gridster2#407]
Hello all.
I have difficulties adding several items at a time.
Suppose I want to add 100 new items in 1 click. Firstly, the adding works slowly and can break the page. Secondly, I need to know if there is enough space to add items. If yes - add. If not, return it back. I check this with itemComponent.notPlaced inside initCallback.
But initCallback works too late, after passing through all the elements. Is there any way to break a cycle of adding if one of the elements returned itemComponent.notPlaced = true?

**Comment by CollinGraf314** [comment]
You can easily add several new items at a time by adding several items to the object holding your GridsterItems. Instead of waiting for the initCallback each time before placing the next item, place them all at once and if any of the initCallbacks reveal that an item wasn't placed, then revert the object holding your GridsterItems back to the state it was in before you added anything to it.

**Comment by ClleaR** [comment]
@CollinGraf314 that's what i'm doing, but as i said before - this operation can break browser page and with lot items it can freeze page for 5-10 sec (i'm also have some heavy calculations there).
So I'm searching the way to avoid that freeze by breaking adding loop or something like that.

**Comment by CollinGraf314** [comment]
There is no built in way to do what you want. You could try creating a global flag `breakAdd: boolean = false;` and then wrap your add-item logic in a conditional if (!breakAdd) and then if at any time in initCallback you see that itemComponent.notPlaced = true set breakAdd to true and stop any more logic from being run.

**Comment by ClleaR** [comment]
@CollinGraf314 yeah i'm tried that before, but setting breakAdd to true works only after finishing whole add loop, so it can't be break by this logic.

**Comment by CollinGraf314** [comment]
You could try using
`getNextPossiblePosition(item: GridsterItem): Boolean  - call to get a viable position for item. Returns true if found.`
before adding each of your items.
